capa/features/extractors/dotnet/extractor.py

```diff
@@ -0,0 +1,61 @@
+from __future__ import annotations
+
+from typing import TYPE_CHECKING, Any, List, Tuple
+
+if TYPE_CHECKING:
+    from capa.features.common import Feature
+
+import dnfile
+
+import capa.features.extractors
+import capa.features.extractors.dotnet.file
+import capa.features.extractors.dotnet.insn
+from capa.features.extractors.base_extractor import FeatureExtractor
+from capa.features.extractors.dotnet.helpers import get_dotnet_methods
+
+
+class DnfileFeatureExtractor(FeatureExtractor):
+    def __init__(self, path: str):
+        super(DnfileFeatureExtractor, self).__init__()
+        self.pe: dnfile.dnPE = dnfile.dnPE(path)
+
+        # pre-compute these because we'll yield them at *every* scope.
+        self.global_features: List[Tuple[Feature, int]] = []
+        self.global_features.extend(capa.features.extractors.dnfile_.extract_file_os(pe=self.pe))
+        self.global_features.extend(capa.features.extractors.dnfile_.extract_file_arch(pe=self.pe))
+
+    def get_base_address(self):
+        return 0x0
+
+    def extract_global_features(self):
+        yield from self.global_features
+
+    def extract_file_features(self):
+        yield from capa.features.extractors.dotnet.file.extract_features(self.pe)
+
+    def get_functions(self):
+        ctx = {}
+        ctx["pe"] = self.pe
+
+        for f in get_dotnet_methods(self.pe):
+            setattr(f, "ctx", ctx)
+            yield f
```

Comment:
I think this works ok. Alternatively, can use the `FunctionHandle` pattern like here: https://github.com/mandiant/capa/blob/54aa14c4f559d4833f17d33abe19615a262c7903/capa/features/extractors/viv/extractor.py#L24

Reply:
I borrowed this pattern from here: https://github.com/mandiant/capa/blob/b455b67da322b28214fbe3590225843728fe4173/capa/features/extractors/ida/extractor.py#L78-L85

we use `ctx` to cache dotnet un/managed imports across functions. I can add comments to make this clearer.